runtime/minor_gc.c

```diff
@@ -337,7 +337,8 @@ static void oldify_one (void* st_v, value v, value *p)
       ft = Tag_val (get_header_val(f) == 0 ? Field(f, 0) : f);
     }
 
-    if (ft == Forward_tag || ft == Lazy_tag || ft == Double_tag) {
+    if (ft == Forward_tag || ft == Lazy_tag ||
+        ft == Forcing_tag || ft == Double_tag) {
```

Comment:
Nice catch! There is similar logic in `runtime/{extern,major_gc,weak}.c` which needs to be fixed in the same way.

Reply:
Done.